frontend/controller/actions/mailbox.js

```diff
@@ -58,26 +58,28 @@ export default (sbp('sbp/selectors/register', {
         data: { username: rootState.loggedIn.username }
       })
 
-      const paramsData = {
-        username: params.data.username,
-        contractID: message.contractID()
-      }
-      await sbp('chelonia/out/actionEncrypted', {
-        ...omit(params, ['options', 'data', 'hook']),
-        data: paramsData,
-        action: 'gi.contracts/mailbox/createDirectMessage'
-      })
-
       await sbp('gi.actions/chatroom/join', {
         ...omit(params, ['options', 'data', 'hook']),
         contractID: message.contractID(),
         data: { username: partnerProfile.username }
       })
 
+      await sbp('chelonia/out/actionEncrypted', {
+        ...omit(params, ['options', 'data', 'hook']),
+        data: {
+          username: params.data.username,
+          contractID: message.contractID()
+        },
+        action: 'gi.contracts/mailbox/createDirectMessage'
+      })
```

Comment:
What `contractID` is this message being sent to? It doesn't seem to be specified, unless it's part of `params`? (But I don't see `'gi.actions/mailbox/createDirectMessage'` called anywhere in the code!)

Reply:
> BTW, I don't see 'gi.actions/mailbox/createDirectMessage' called anywhere in the code! Why is that?

That's because the action name is set dynamically according to the existence of direct message.
`Line #165 in ProfileCard.vue`
`Line #182 in NewDirectMessageModal.vue`

> What contractID is this message being sent to? It doesn't seem to be specified, unless it's part of params?

If you check those two lines above, you can see the contractID there.